### PR TITLE
Form 0781/0781a: Section Headings on Summary Page and page reordering #15757

### PR DESCRIPTION
**Pull request by zurbergram**
## Description

As a va.gov developer I want to ensure that the 0781/0781a sections are intuitively named so that when the user views the final summary the section headings make sense.

## Testing done
- [x] local testing

## Screenshots
[Apply for disability benefits_ VA.gov.pdf](https://github.com/department-of-veterans-affairs/vets-website/files/2691233/Apply.for.disability.benefits_.VA.gov.pdf)

## Acceptance criteria
- 'Unknown Condition' is replaced with 'Factors that contributed to PTSD'

For Form 0781:
- 'PTSD Walkthrough 781 Choice' is replaced with 'Answer online questions or upload paper 21-0781'
- 'First Medals or citations' is replaced with 'Medals or citations associated with [first] event'
- 'First PTSD incident date' is replaced with 'Date of [first] event'
- 'First PTSD incident unit assignment' is replaced with 'Unit assignment for [first] event'
- 'First PTSD incident location' is replaced with 'Location of [first] event'
- 'First PTSD individuals involved yes/no' is replaced with 'Were other people involved in the [first] event?'
- 'First 781 PTSD Event Description' is replaced with 'Description of [first] event'
- 'First PTSD incident Additional events.' is replaced with 'Add another event or situation?'
- 'Additional Remarks- 781' is replaced with 'Additional Remarks'

For Form 0781a:
- 'PTSD Walkthrough 781a Choice' is replaced with 'Answer online questions or upload paper 21-0781A?
- 'First PTSD assault incident date' is replaced with 'Date of [first] event'
- 'First 781a PTSD Event Description' is replaced with 'Description of [first] event'
- 'First PTSD assault incident unit assignment' is replaced with 'Unit assignment for [first] event'
- 'First PTSD assault incident location' is replaced with 'Location of [first] event'
- 'First PTSD assault authorities' is replaced with 'Reports from authorities'
- 'First PTSD assault additional events' is replaced with 'Add another event or situation?'
- 'Additional Remarks- Physical Health Changes' is replaced with 'Additional changes in behavior- physical'
- Should have section for: 'Additional changes in behavior- mental/substance abuse'
- Should have section for: 'Additional changes in behavior- work'
- Should have section for: 'Additional changes in behavior- social'
- 'Additional Remarks- Additional Behavior Changes' is replaced with 'Additional changes in behavior- more information'

## Definition of done
- [x] Events are logged appropriately
- [x] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

**Review by aaronjlech — APPROVED** [review]
lgtm

**Comment by zurbergram** [comment]
@department-of-veterans-affairs/code-reviewers-for-external-teams-group - Could I get a code review on this at your earliest convenience? Thanks!

File: src/applications/disability-benefits/all-claims/config/781/index.js [hunk 1]
```diff
@@ -39,17 +39,40 @@ const numberToWords = {
...
+  }Events${additionalIncidentKeyIndex}`;
 
   let formattedTitle = title;
 
   if (formData[additionalIncidentKey]) {
-    formattedTitle = `${numberToWords[index]} ${title}`;
+    if (title.search(REVIEW_TITLE_TOKEN) > 0) {
+      formattedTitle = title.replace(
+        REVIEW_TITLE_TOKEN,
+        ` ${numberToWords[index].toLowerCase()} `,
```
Comment (erikphansen) [inline src/applications/disability-benefits/all-claims/config/781/index.js]: Is is possible for this index to be > 9 (longer than the `numberToWords` array)?

File: src/applications/disability-benefits/all-claims/config/781/index.js [hunk 2]
```diff
@@ -39,17 +39,40 @@ const numberToWords = {
...
+      );
+    } else {
+      // If does not contain REVIEW_TITLE_TOKEN put numberToWords[index] at start of title
+      formattedTitle = `${numberToWords[index]} ${title}`;
```
Comment (erikphansen) [inline src/applications/disability-benefits/all-claims/config/781/index.js]: Same ? here.

File: src/applications/disability-benefits/all-claims/config/781/index.js [hunk 3]
```diff
@@ -39,17 +39,40 @@ const numberToWords = {
...
+/**
+ * This removes "First " from the title if there is only one incident.
+ *
+ * @param {string} [title] Displayed as the section summary header.
+ * If contains REVIEW_TITLE_TOKEN and there is more than one incident, replaces REVIEW_TITLE_TOKEN with numberToWords[index].toLowerCase().
+ * If does not contain REVIEW_TITLE_TOKEN appends numberToWords[index] to front of title.
+ * @param {int} index Index of numberToWords
+ * @param {string} formType Indicates what type of form is calling function; 781, 781a
+ * @returns {object} title
+ */
```
Comment (erikphansen) [inline src/applications/disability-benefits/all-claims/config/781/index.js]: This whole function is begging for some unit tests.

**Review by erikphansen — APPROVED** [review]
LGTM, but it should probably also be verified by someone with more knowledge of 526.

File: src/applications/disability-benefits/all-claims/config/781/index.js (hunk 1 above)
Comment (zurbergram) [reply src/applications/disability-benefits/all-claims/config/781/index.js]: can't have more than 3 additional events during the 781/a flow

File: src/applications/disability-benefits/all-claims/config/781/index.js (hunk 2 above)
Comment (zurbergram) [reply src/applications/disability-benefits/all-claims/config/781/index.js]: can't have more than 3 additional events during the 781/a flow

**Pull request merged**